integration-cli/docker_cli_pull_test.go

```diff
@@ -59,28 +59,36 @@ func TestPullVerified(t *testing.T) {
 	// unless keychain is manually updated to contain the daemon's sign key.
 
 	verifiedName := "hello-world"
+	expected := "The image you are pulling has been verified"
+	untargeted := "v1"
+
 	defer deleteImages(verifiedName)
 
 	// pull it
-	expected := "The image you are pulling has been verified"
 	pullCmd := exec.Command(dockerBinary, "pull", verifiedName)
-	if out, exitCode, err := runCommandWithOutput(pullCmd); err != nil || !strings.Contains(out, expected) {
-		if err != nil || exitCode != 0 {
-			t.Skipf("pulling the '%s' image from the registry has failed: %s", verifiedName, err)
-		}
-		t.Fatalf("pulling a verified image failed. expected: %s\ngot: %s, %v", expected, out, err)
+	out, exitCode, err := runCommandWithOutput(pullCmd)
+	if err != nil || exitCode != 0 {
+		t.Skipf("pulling the '%s' image from the registry has failed: %s", verifiedName, err)
 	}
 
-	// pull it again
-	pullCmd = exec.Command(dockerBinary, "pull", verifiedName)
-	if out, exitCode, err := runCommandWithOutput(pullCmd); err != nil || strings.Contains(out, expected) {
-		if err != nil || exitCode != 0 {
-			t.Skipf("pulling the '%s' image from the registry has failed: %s", verifiedName, err)
+	if strings.Contains(out, untargeted) {
+		logDone("pulling from v1 registry, no need verified")
```

Comment:
I think that statement is generally true.  However since we do not do verification on v1 images, this test does apply here. Not getting an image from the v2 registry means it cannot be verified and therefore cannot pass, it either needs to be treated as an error or skipped. Eventually pulling official images from a v1 registry will be disallowed and this skip will changed to fatal.  @duglin do you agree in this case the skip semantics would apply?